### PR TITLE
Pontusfilter

### PR DESCRIPTION
**Pull request by 3xer**
Added live text search that ignores uppercase and or lowercase letters for more ease of use

### CODE REVIEW

File: challenges.html [hunk 1]
```diff
@@ -97,7 +97,7 @@ <h3 class="h3-search-keyword">Or type to search for keyword</h3>
```
Comment (3xer) [inline challenges.html:97]: Remove changes to this file they're not relevant

File: textFilter.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+async function displayRooms() {
+    const res = await fetch("https://lernia-sjj-assignments.vercel.app/api/challenges%27");
+    const data = await res.json();
+    data.challenges.forEach(challenge => {
+        console.log(challenge.title)
+    });
+}
+
+searchInput.addEventListener("input", function(){
+    sl = searchInput.innerHTML.length;
```
Comment (rfabricioflores) [inline textFilter.js]: Här kanske ska man ändra innerHTML för innerText eftersom innerHTML läser alla html taggar också och det kan bli ett problem. Du kan testa

File: textFilter.js [hunk 3]
```diff
@@ -1,34 +1,141 @@
...
+            content.innerText = "\nTitle: " + challenge.title + "\nDescription: " + challenge.description;
+            item.appendChild(content);
+            roomsUl.appendChild(item);
+       }
     });
+
+    
+}
+
+    async function APIGet(){
```
Comment (rfabricioflores) [inline textFilter.js]: APIGet() kan man ändra till getChallenges()
För att göra koden mer läsbart

**Review by Arash-SJPI22 — COMMENTED** [review]
Looks good, but would be great if you could put the search function in a seperate file.

File: textFilter.js [hunk 4]
```diff
@@ -1,44 +1,91 @@
...
     function search(){
-        
+       
+
         const searchString = searchInput.value;
-        console.log(searchString);
-        render(searchInput.value);
+       
+        //if string empty display all rooms
+        if(searchString == ''){
```
Comment (rfabricioflores) [inline textFilter.js]: Det går också att göra string.length <= 0;

File: textFilter.js [hunk 5]
```diff
@@ -1,188 +1,69 @@
...
+// renders all rooms if the input (searchString) is empty
+async function renderAll(){
+    const data = await APIGet();
+    roomsUl.innerText = '';
+    data.challenges.forEach(challenge => {
+        const roomsUl = document.querySelector(".rooms");
+        const item = document.createElement("li");
+        const content = document.createElement("p");
+
+        content.innerText = "\nTitle: " + challenge.title + "\nDescription: " + challenge.description;
```
Comment (rfabricioflores) [inline textFilter.js:24]: Använd interpolation (det är oftast enklare)
ex. `Title ${challenge.title}`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: textFilter.js (hunk 2 above)
Comment (3xer) [reply textFilter.js]: It's solved the code if from an older commit :)

File: textFilter.js (hunk 5 above)
Comment (3xer) [reply textFilter.js:24]: Good idea, we can use that for the proper render Vedad made if he hasn't already or if he did it in another way.

File: textFilter.js (hunk 4 above)
Comment (3xer) [reply textFilter.js]: 👍

File: textFilter.js (hunk 3 above)
Comment (3xer) [reply textFilter.js]: I'm thinking that API in the name is good for understanding what it is, but APIGet may not be the one to use.

**Comment by Arash-SJPI22** [comment]
Not approved, re-request with changes

**Pull request closed without merging**